controllers/pod_client.go

```diff
@@ -143,30 +146,39 @@ func (client *realFdbPodClient) getListenIP() string {
 
 // makeRequest submits a request to the sidecar.
 func (client *realFdbPodClient) makeRequest(method string, path string) (string, error) {
-	var protocol string
-	if client.useTLS {
-		protocol = "https"
-	} else {
-		protocol = "http"
-	}
-
-	url := fmt.Sprintf("%s://%s:8080/%s", protocol, client.getListenIP(), path)
 	var resp *http.Response
 	var err error
 
-	httpClient := &http.Client{}
+	protocol := "http"
+	retryClient := retryablehttp.NewClient()
+	retryClient.RetryMax = 10
+	retryClient.RetryWaitMax = 5 * time.Second
+	// Prevent logging
+	retryClient.Logger = nil
+	retryClient.CheckRetry = func(c context.Context, resp *http.Response, err error) (bool, error) {
```

Comment:
If we change the way how we invoke the `CheckHash` we could leverage this to actually check multiple times with an exponential backoff.